Gems/Terrain/Code/Source/TerrainRenderer/TerrainDetailMaterialManager.cpp

```diff
@@ -764,20 +777,29 @@ namespace Terrain
                             pixels.at(index).m_material1 = aznumeric_cast<uint8_t>(materialId);
                             firstWeight = surfaceTagWeight.m_weight;
                             isFirstMaterial = false;
-                            static constexpr float MaxValueBeforeRounding = 254.5f / 255.0f;
-                            if (firstWeight >= MaxValueBeforeRounding)
-                            {
-                                break;
-                            }
                         }
                         else
                         {
                             // Second material is valid, weight is relative based on first material's weight.
-                            pixels.at(index).m_material2 = aznumeric_cast<uint8_t>(materialId);
-                            float totalWeight = firstWeight + surfaceTagWeight.m_weight;
-                            float blendWeight = 1.0f - (firstWeight / totalWeight);
-                            pixels.at(index).m_blend = aznumeric_cast<uint8_t>(AZStd::round(blendWeight * 255.0f));
-                            break;
+                            if (surfaceTagWeight.m_weight < 1.0 - firstWeight)
```

Comment:
Are these surfacetagWeigths in surfacePoint.m_surfaceTags always pre-sorted? 

Also, as someone unfamiliar with how this all comes together, it's not immediately clear to me why 1.0 - firstWeight is the weight of the default material.

Reply:
They are presorted yes, so the first weight is always the highest. If the highest weight is less than 0.5, then you know the second weight will be even less, so this is saying to use the default material instead in that case.

Reply:
This is actually where some subjectivity comes in and there's different ways to think about it. I tried doing this where it'd only early-out to using the default material when the weight was less than 33% (because a second material could also be 33% leaving 33% for the default material) and only choose to use the default material when 1 minus the weights of the first two materials was less than the second material.... that actually makes more sense to me, but ends up looking slightly worse in my test case. I think some of this is going to come down to content - you really want on any given pixel for there to be at least gradient weight maps (think of them as splat maps) with a weight greater than 50% on that pixel.